tests/Integration/Infrastructure/Persistence/AssetsTableLifecycleTest.php

```
@@ -0,0 +1,161 @@
+                ],
+                $this->fetchPersistedLifecycleState($connection, (string) $row['upload_id']),
+            );
+        });
+    }
+
+    /**
+     * @param array<string,int|string|null> $overrides
+     */
+    #[Test]
+    #[DataProvider('provideInvalidLifecycleCases')]
+    public function itThrowsConstraintViolationWhenRowInvalid(array $overrides, string $message, ?string $baseRowName = null, ?string $expectedSqlState = '23000', ?string $expectedMessageFragment = null, bool $preInsertBaseRow = false): void
```

PHPMD (2.15.0)

[error] The method itThrowsConstraintViolationWhenRowInvalid has a boolean flag argument $preInsertBaseRow, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)